neo/core/basesignal.py

```diff
@@ -303,3 +303,65 @@ def merge(self, other):
             signal.channel_index = ChannelIndex(index=np.arange(signal.shape[1]))
 
         return signal
+
+    @staticmethod
+    def _rescale_epoch_times(a_signal, times_of_an_epoch):
+        """
+        Checks epoch.times.units against signal.times.units
+        
+        Arguments:
+        a_signal
+        times_of_an_epoch; epc.times or epc.durations of the created epoch, epc.
+
+        Returns:
+        same times_of_an_epoch if units are the same
+        or
+        rescaled (to signal.times.units) times_of_an_epoch if units are different.
+        """
+        if times_of_an_epoch.units is not a_signal.times.units:
+            return times_of_an_epoch.rescale(a_signal.times.units)
+        else:
+            return times_of_an_epoch
+
+    def extract_for_epoch(self, epoch):
```

Comment:
maybe `cut_by_epoch` or `slice_by_epoch` is more telling?

Reply:
Hi Julia,
Thanks for the comments.
The point regarding the function performing a utility is a good one. I started out thinking that way but since there were no current `utility.py` script/source fille I took the current approach which is to make signal have the method where one just had to pass the epoch. If there's going to be a script containing the utility methods I thinks its ok to have it there.

With regards to why I have `_rescale_epoch_time` and `extract_for_epoch`, not just one is because it avoids code duplication and also one has only 1-reason to change/edit either methods.

Reply:
I prefer `slice_by_epoch` to match `time_slice`

Reply:
I agree with @JuliaSprenger on moving `_rescale_epoch_time` inside the function where it is used. This does not imply code duplication, since you can define a function inside a function.

Alternatively, it could be moved out of the BaseSignal definition into a utility function, as checking two Neo objects have the same units must surely be a common need.